### PR TITLE
Improve comments for surctionproof init+alloc/destroy funcs

### PR DESCRIPTION
**Pull request by dgpv**
Improve comments for surctionproof init+alloc/destroy funcs
    
    The comments with 'XXX' was intended to indicate that the listed
    concerns was subject to review and change, but the code with these
    comments was merged straight away. This commit replaces comments
    with more complete text describing the issues.
    
    This also signifies that the commit that this code was introduced in is
    not anymore 'work in progress'.

### CODE REVIEW

**Comment by dgpv** [comment]
It seems important to me to remove that to avoid potential confusion for people that will read the code in the future

**Comment by apoelstra** [comment]
Sounds good! I looked at these comments when merging the old PR and I thought they were fine to keep in for some long-future improvements, but I have no strong feelings about it.

**Comment by dgpv** [comment]
At least the comments should have more information and no 'XXX' tag. Let me prepare better version for the comments.

**Comment by dgpv** [comment]
Added detailed comments describing the issues concerned, instead.

**Comment by apoelstra** [comment]
Sounds good. Honestly I think it's a bit too wordy, but again, I don't have strong feelings :)

**Comment by dgpv** [comment]
ACK
